ImageFeed/Scenes/ImagesList/SingleImageView/SingleImageViewController.swift

```diff
@@ -0,0 +1,216 @@
+
+        NSLayoutConstraint.activate([
+            scrollView.topAnchor.constraint(
+                equalTo: view.topAnchor),
+            scrollView.leadingAnchor.constraint(
+                equalTo: view.leadingAnchor),
+            scrollView.trailingAnchor.constraint(
+                equalTo: view.trailingAnchor),
+            scrollView.bottomAnchor.constraint(
+                equalTo: view.bottomAnchor)
+        ])
+    }
+
+    private func layoutImageView() {
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+
+        scrollView.addSubview(imageView)
+
+        let scrollGuide = scrollView.contentLayoutGuide
+
+        NSLayoutConstraint.activate([
+            imageView.topAnchor.constraint(
+                equalTo: scrollGuide.topAnchor),
+            imageView.leadingAnchor.constraint(
+                equalTo: scrollGuide.leadingAnchor),
+            imageView.trailingAnchor.constraint(
+                equalTo: scrollGuide.trailingAnchor),
+            imageView.bottomAnchor.constraint(
+                equalTo: scrollGuide.bottomAnchor)
+        ])
+    }
+
+    private func layoutBackButton() {
+        backButton.translatesAutoresizingMaskIntoConstraints = false
+
+        view.addSubview(backButton)
+
+        let safeArea = view.safeAreaLayoutGuide
+
+        NSLayoutConstraint.activate([
+            backButton.topAnchor.constraint(
+                equalTo: safeArea.topAnchor, constant: 9),
+            backButton.leadingAnchor.constraint(
+                equalTo: safeArea.leadingAnchor, constant: 9)
+        ])
+    }
+
+    private func layoutShareButton() {
+        shareButton.translatesAutoresizingMaskIntoConstraints = false
+
+        view.addSubview(shareButton)
+
+        let safeArea = view.safeAreaLayoutGuide
+
+        NSLayoutConstraint.activate([
+            shareButton.centerXAnchor.constraint(
+                equalTo: safeArea.centerXAnchor),
+            shareButton.bottomAnchor.constraint(
+                equalTo: safeArea.bottomAnchor, constant: -16)
+        ])
+    }
+}
+
+// MARK: - Styling
+
+extension SingleImageViewController {
+    private func setupView() {
+        view.backgroundColor = .asset(.ypBlack)
+    }
+}
+
+// MARK: - UIScrollViewDelegate
+
+extension SingleImageViewController: UIScrollViewDelegate {
+    func viewForZooming(in scrollView: UIScrollView) -> UIView? {
+        imageView
+    }
+
+    private func rescaleAndCenterImageInScrollView(image: UIImage) {
+        let minZoomScale = scrollView.minimumZoomScale
+        let maxZoomScale = scrollView.maximumZoomScale
+
+        let containerSize = view.bounds.size
+        let imageSize = image.size
+
+        let hScale = containerSize.width / imageSize.width
+        let vScale = containerSize.height / imageSize.height
+
+        let scale = min(maxZoomScale,
+                        max(
+                            minZoomScale,
+                            max(hScale, vScale)))
```

Comment:
Если используете форматирование, где первый параметр на той же строке, что и функция, то тогда уже последовательно:
```suggestion
        let scale = min(maxZoomScale,
                        max(minZoomScale,
                            max(hScale, vScale)))
```

но имхо, minmax's обычно в одну строку пишут

Reply:
Да, действительно странно вышло, у меня просто голова от этого метода немного распухла, вот я и пытался себе пояснить :) Если честно, то в одно строчку не так и страшно выглядит! Все что угодно лучше чем мое форматирование, Спасибо за замечание